### ISSUE TITLE
Removed lint missing_return breaks customer tests, will not roll into framework

### ISSUE THREAD

**Issue by whesse** [flutter/flutter#141576]
Rolls of the current tip of Dart SDK into framework will be blocked until this is resolved

The lint missing_return is removed from the Dart analyzer by https://dart-review.googlesource.com/c/sdk/+/346182

An analysis_options.yaml file that contains that lint will now cause the analyzer to produce a warning when run.

This causes a failure in customer_tests, and may cause other failures we don't see yet in tests that aren't run on monorepo.
The failures in customer_tests are fixed by two PRs, one already landed:
 * (landed) https://github.com/flutter/devtools/pull/7045
 * https://github.com/marcglasberg/assorted_layout_widgets/pull/31
 
 If these upstream fixes land, then the roll to framework may succeed.

**Comment by whesse** [comment]
This issue should now be fixed. The fix will be tested first on the next build of the monorepo customer_tests builder: https://ci.chromium.org/ui/p/dart/builders/luci.dart.ci.sandbox/flutter-linux-customer-testing

**Comment by zanderso** [comment]
Still failing to roll to the engine https://github.com/flutter/flutter/pull/141634. I reverted the Dart SDK -> Engine roll, and stopped the autoroller.

**Comment by zanderso** [comment]
cc @scheglov

**Comment by scheglov** [comment]
Can you update `analysis_options.yaml` files to remove references to `missing_return`?

**Comment by zanderso** [comment]
Generally speaking, it's the responsibility of the person or team making the breaking change to perform the migration. The responsibility of the engine sheriff is only to get the tree green, rollers unblocked, and issues filed. @scheglov if you're looking for help in doing the migration, the best way to go about that is to coordinate with the appropriate team before landing the breaking change(s).

**Comment by scheglov** [comment]
Is there a way to see which packages are affected?

**Comment by zanderso** [comment]
I believe @whesse can advise on the places in Dart CI to check for failures. The affected packages would have to be inferred from that. Alternately, grepping in the framework repo for the deprecated strings would probably be a good starting point.

**Comment by scheglov** [comment]
Grepping like [this](https://github.com/search?q=org%3Aflutter+%22missing_return%3A%22&type=code)?
Or in other locations than `org:flutter`?

**Comment by goderbauer** [comment]
Note from triage: @scheglov I assigned this to you since you seem to be looking into this.

**Comment by scheglov** [comment]
OK, I can update the analyzer to stop reporting this warning for `missing_return`.
Then the clients can update their `analysis_options.yaml` files at the pace that it convenient to them.
https://dart-review.googlesource.com/c/sdk/+/346681

**Comment by scheglov** [comment]
The CL landed.
Let me know if there are still failures (and [where to find them](https://en.wikipedia.org/wiki/Fantastic_Beasts_and_Where_to_Find_Them_(film))).

**Comment by zanderso** [comment]
It looks like there are still some tests failing https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20tool_integration_tests_4_4/31150/overview

**Comment by zanderso** [comment]
I understand what you're saying, however as I said over chat, the team-friendly way to approach changes like this is to revert first, and follow up with the team that owns the broken test outside the context of a P0 roll-breakage to figure out the best path to migrate. If you'd like this migration to be addressed as a P0, then it makes sense to me that you would be the one to do the cleanup that that implies.

**Comment by scheglov** [comment]
OK, I opened a [new issue](https://github.com/flutter/flutter/issues/141743).
Is this the right location?
Do I need to set any labels?
Hopefully the issue will get attention.

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/346948 will revert a 2 related CLs.

**Comment by scheglov** [comment]
The revert landed.

**Comment by zanderso** [comment]
> The revert landed.

Thanks!

**Comment by a-siva** [comment]
Closing as a revert has landed.

**Comment by scheglov** [comment]
@whesse Is there a way for me to avoid such roll issues in the future?
Is there a bot that I can trigger on CL to verify that it will not cause issues for Flutter, and whatever else is checked during SDK roll?

**Comment by a-siva** [comment]
> @whesse Is there a way for me to avoid such roll issues in the future? Is there a bot that I can trigger on CL to verify that it will not cause issues for Flutter, and whatever else is checked during SDK roll?

I was told that this failure did show up in the HHH bot that we have here https://ci.chromium.org/p/dart/g/flutter-engine/console, is that not the case ?

**Comment by scheglov** [comment]
HHH is too late? I'd like to know this in advance, before my CL lands, as a pre-submit.

**Comment by whesse** [comment]
Yes, there is a try job, flutter-linux-try, that will run many framework tests against your Gerrit CL. There is also flutter-web-try, that runs the flutter web tests. The framework tests should definitely be run against CLs that will change the analyzer or linter, and also for DEPS changes.

I think we fixed all the breakages visible on those try  builders, but this issue lists a Flutter CI failure that occurred when this last landed.  Those should be checked, and fixed. The tests can be checked in a monorepo checkout, by building and running local tests, but in this case, the fixes are easy, I think.

The failure is at https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20tool_integration_tests_4_4/31150/overview. If you haven't landed a PR to fix it, it will still be broken.

It can be tricky to run the framework sharded tests locally with a local_engine setting,  They can sometimes try and use the pinned SDK downloaded by the framework, rather than the locally built one, if the right flags aren't used.

**Comment by whesse** [comment]
OK, I see a PR that fixed it has landed: https://github.com/flutter/flutter/pull/141790.

You can manually do these tests that the PR has changed, but since your CL does not affect the "unused_element" lint, your CL should not be in danger of breaking these tests any more.  Since the tool_integration tests are the only ones that failed on the last attempt to roll your CL to framework (see the "flutter dashboard" checks at https://github.com/flutter/flutter/pull/141634/checks), I don't think you need to do any further testing.

**Comment by scheglov** [comment]
I tried to run `flutter-linux-try` and `flutter-web-try` bots, see https://dart-review.googlesource.com/c/sdk/+/348185
But `flutter-web-try` fails.
And `flutter-linux-android_debug-try`.
The failures are violet, indicating infra issues?
So, it is not safe to land, because we don't know whether it is green or red?
Is it a known issue?

**Comment by whesse** [comment]
It is safe to land. Yes, this is an infra issue, the source checkout on the debug android builder timed out after an hour.
The flutter-web-try just summarizes the builds it triggers, including this failure on debug android. Running these builders is optional, so if they fail, it is up to you if you retry or just ignore them.  

The original CL didn't cause any failures on flutter-web, and running this builder is just advisory. The status quo is still that SDK changes that don't cause failures in our CI will have failures detected when they are rolled into Flutter, and the roll sheriffs will try and fix or roll back.  We just have some better tools now to find these failures out early, before they land and roll, and so we can use these tools when they are worth it.
